vyper/venom/passes/branch_optimization.py

```diff
@@ -0,0 +1,32 @@
+from vyper.venom.analysis.cfg import CFGAnalysis
+from vyper.venom.analysis.dfg import DFGAnalysis
+from vyper.venom.passes.base_pass import IRPass
+
+
+class BranchOptimizationPass(IRPass):
+    """
+    This pass optimizes branches inverting jnz instructions where appropriate
+    """
+
+    def _optimize_branches(self) -> None:
+        fn = self.function
+        for bb in fn.get_basic_blocks():
+            term_inst = bb.instructions[-1]
+            if term_inst.opcode != "jnz":
+                continue
+
+            prev_inst = self.dfg.get_producing_instruction(term_inst.operands[0])
+            if prev_inst.opcode == "iszero":
+                prev_inst.parent.remove_instruction(prev_inst)
```

Comment:
i think this could be wrong if prev_inst.output is still live

Reply:
new version does not remove instructions